package/gluon-respondd/luasrc/lib/gluon/upgrade/400-respondd-firewall

```diff
@@ -6,26 +6,46 @@ uci:delete('firewall', 'wan_announced')
 
 -- Allow respondd port on WAN to allow resolving neighbours over mesh-on-wan
 uci:section('firewall', 'rule', 'wan_respondd',
-  {
-    name = 'wan_respondd',
-    src = 'wan',
-    src_ip = 'fe80::/64',
-    dest_port = '1001',
-    proto = 'udp',
-    target = 'ACCEPT',
-  }
+{
+	name = 'wan_respondd',
+	src = 'wan',
+	src_ip = 'fe80::/64',
+	dest_port = '1001',
+	proto = 'udp',
+	target = 'ACCEPT',
+}
 )
 
 -- Restrict respondd queries to link-local addresses to prevent amplification attacks from outside
 uci:section('firewall', 'rule', 'client_respondd',
-  {
-    name = 'client_respondd',
-    src = 'client',
-    src_ip = '!fe80::/64',
-    dest_port = '1001',
-    proto = 'udp',
-    target = 'REJECT',
-  }
+{
+	name = 'client_respondd',
+	src = 'client',
+	src_ip = '!fe80::/64',
+	dest_port = '1001',
+	proto = 'udp',
+	target = 'REJECT',
+}
 )
 
+uci:section('firewall', 'rule',  'mesh_respondd_mcast1',
+{
+	name = 'mesh_respondd',
+	src = 'mesh',
+	src_ip = site.prefix6,
+	dest_port = '1001',
+	proto = 'udp',
+	target = 'ACCEPT',
+}
+)
+uci:section('firewall', 'rule',  'mesh_respondd_ll',
+{
+	name = 'mesh_respondd',
+	src = 'mesh',
+	src_ip = 'fe80::/64',
+	dest_port = '1001',
+	proto = 'udp',
+	target = 'ACCEPT',
+}
+)
```

Comment:
I do agree that the current commit structure needs improvement. That being said, splitting firewall changes like that into multiple commits means that I'd have to re-write rules for each specific commit to make the whole firewall sound for each commit. 
This is quite a bit of work with little benefit. I feel that the structural changes should be a commit and then additional rules (like  allowing access to respondd  from additions ip-addresses) could be in separate commits.